cmd/heimdalld/init.go

```diff
@@ -37,25 +37,41 @@ type initHeimdallConfig struct {
 }
 
 func heimdallInit(_ *server.Context, cdc *codec.Codec, initConfig *initHeimdallConfig, config *cfg.Config) error {
-	// do not execute init if forceInit is false and genesis.json already exists (or we do not have permission to write to file)
-	if !initConfig.forceInit {
-		_, err := os.Stat(config.GenesisFile())
-		if err == nil || !errors.Is(err, os.ErrNotExist) { // https://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go
-			return nil
-		}
-	}
-
 	WriteDefaultHeimdallConfig(filepath.Join(config.RootDir, "config/heimdall-config.toml"), helper.GetDefaultHeimdallConfig())
 
 	nodeID, valPubKey, _, err := InitializeNodeValidatorFiles(config)
 	if err != nil {
 		return err
 	}
 
-	genesisCreated, err := helper.WriteGenesisFile(initConfig.chain, config.GenesisFile(), cdc)
-	if err != nil {
-		return err
-	} else if genesisCreated {
+	// do not execute init if forceInit is false and genesis.json already exists (or we do not have permission to write to file)
+	writeGenesis := initConfig.forceInit
+
+	if !writeGenesis {
+		// When not forcing, check if genesis file exists
+		_, err := os.Stat(config.GenesisFile())
+		if err != nil && errors.Is(err, os.ErrNotExist) { // https://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go
```

Comment:
Do we need this StackOverflow link in the comment?

Reply:
Yes, I think we can delete this line :D Will delete it.